Frontend/ui-driver/src/Screens/ReferralScreen/View.purs

```diff
@@ -67,34 +70,50 @@ screen initialState =
-                      case leaderBoardRes of
-                        Right res -> lift $ lift $ doAff do liftEffect $ push $ UpdateLeaderBoard res
-                        Left err  -> lift $ lift $ doAff do liftEffect $ push $ UpdateLeaderBoardFailed
-                      pure unit
+                            leaderBoardRes <- lift $ lift $ Remote.leaderBoard $ DailyRequest (convertUTCtoISC selectedDay.utcDate "YYYY-MM-DD")
+                            case leaderBoardRes of
+                              Right res -> lift $ lift $ doAff do liftEffect $ push $ UpdateLeaderBoard res
+                              Left err  -> lift $ lift $ doAff do liftEffect $ push $ UpdateLeaderBoardFailed
+                            pure unit
+                          ST.Weekly -> do
+                            let selectedWeek =  if any (_ == "") [initialState.props.selectedWeek.utcStartDate, initialState.props.selectedWeek.utcEndDate] then
+                                                  case last (getPastWeeks 1) of
+                                                    Just week -> week
+                                                    Nothing -> initialState.props.selectedWeek
+                                                else initialState.props.selectedWeek
+                            leaderBoardRes <- lift $ lift $ Remote.leaderBoard $ WeeklyRequest (convertUTCtoISC selectedWeek.utcStartDate "YYYY-MM-DD") (convertUTCtoISC selectedWeek.utcEndDate "YYYY-MM-DD")
+                            case leaderBoardRes of
+                              Right res -> lift $ lift $ doAff do liftEffect $ push $ UpdateLeaderBoard res
+                              Left err  -> lift $ lift $ doAff do liftEffect $ push $ UpdateLeaderBoardFailed
+                            pure unit
+                    ST.QRScreen -> do 
+                      let refCode = getValueToLocalStore REFERRALCODE
+                      (GetPerformanceRes getPerformanceres) <- Remote.getPerformanceBT (GetPerformanceReq {} )
+                      lift $ lift $ doAff do liftEffect $ push $ UpdateDriverPerformance (GetPerformanceRes getPerformanceres)
+                      if (any (_ == refCode) ["__failed", "", "(null)"]) then do
+                          response <- lift $ lift $ Remote.generateReferralCode (GenerateReferralCodeReq {} )
+                          case response of
+                            Right (GenerateReferralCodeRes generateReferralCodeRes) -> lift $ lift $ doAff do liftEffect $ push $ (UpdateReferralCode (GenerateReferralCodeRes generateReferralCodeRes))
+                            Left error -> pure unit
+                      else pure unit 
+                    _ -> pure unit
                 pure (pure unit)
               )
   ]
-  , eval
+  , eval : (\action state -> do
```

Comment:
remove if not required.

Reply:
its required mercy to track action while debug, any way while bundling it will remove all spy's